src/components/ExpiryIndicator/ExpiryIndicator.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { FC, useMemo } from "react";
+import { useTranslation } from "react-i18next";
+
+import { compareAsc, format } from "date-fns";
+
+import {
+  Container,
+  Strong,
+  StyledPill,
+  StyledTooltip,
+} from "./ExpiryIndicator.styles";
+import { getExpiryTranslation } from "./helpers";
+
+type ExpiryIndicatorProps = {
+  expiry: Date;
+};
+
+const ExpiryIndicator: FC<ExpiryIndicatorProps> = ({ expiry }) => {
+  const { t } = useTranslation();
+
+  const hasPassed = useMemo(
+    () => compareAsc(new Date(), expiry) === 1,
+    [expiry]
+  );
+  const timeLeft = useMemo(
+    () => getExpiryTranslation(new Date(), expiry, t),
+    [expiry, t]
+  );
+
+  return (
+    <Container>
+      <StyledPill>
+        {hasPassed ? t("orders.expired") : t("orders.expiresIn")}
+        {!hasPassed && <Strong>{timeLeft}</Strong>}
+      </StyledPill>
+      <StyledTooltip>{format(expiry, "dd-MMM-yyyy pppp")}</StyledTooltip>
```

Comment:
We should display the right format according to the local date of the user. 
I don't no the lib you are using here, but we can do that easely in Javascript with .localeDateString() for the date format + .localeTimeString() for the time format. A concat of both will give your a nice format according of the locale setting from the user. What do you think ?

Reply:
Good point. Just tried using localeDateString and localeTimeString, it works but I have to write an extra helper for getting the timezone info readable. So the best way is still date-fns I think.

Reply:
Yes if you prefer the date-fns libs go a head, i just don't no this lib, i'm use to work with native date object or moment.
If you can display to the user according to his local timezone, it would be verry nice for the user.

Reply:
Date-fns is very similar to moment. So it also takes user's timezone into account.